src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -3836,6 +3850,22 @@ void writeTime(java.sql.Timestamp value, int scale) throws SQLServerException {
         writeScaledTemporal(calendar, subSecondNanos, scale, SSType.TIME);
     }
 
+    void writeTime(java.sql.Timestamp value, int scale, Calendar cal) throws SQLServerException {
+        GregorianCalendar calendar = initializeCalender(TimeZone.getDefault());
+        long utcMillis; // Value to which the calendar is to be set (in milliseconds 1/1/1970 00:00:00 GMT)
+        int subSecondNanos;
+        utcMillis = value.getTime();
+        subSecondNanos = value.getNanos();
```

Comment:
Declarations and assignments can be combined.
```suggestion
        long utcMillis = value.getTime(); // Value to which the calendar is to be set (in milliseconds 1/1/1970 00:00:00 GMT)
        int subSecondNanos = value.getNanos();
```